### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by Naledi-Dikgale**
### In this PR we did the following

- [ ] Removed `current_user` method in `ApplicationController,`

- [ ] Enabled the app to register a new user.

- [ ] User logs in with a combination of email and password.

- [ ] Hashed password is stored in the database.

- [ ] Asks user for confirmation of email.

- [ ] Made it possible for user to reset password.

- [ ] Modify the views in devise.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,6 +1,16 @@
 class ApplicationController < ActionController::Base
-  helper_method :current_user
-  def current_user
-    @current_user ||= User.first
+  # helper_method :current_user
+  # def current_user
+  #   @current_user ||= User.first
+  # end
```
Comment (Lizdev-05) [inline app/controllers/application_controller.rb]: - [ ] Please make sure you remove commented code from your project to make your code clean and tidy :pray: You should not push commented code to your repository. You can read more about it [here](https://www.freecodecamp.org/news/code-comments-the-good-the-bad-and-the-ugly-be9cc65fbf83/)

File: app/models/user.rb [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
   has_many :posts, foreign_key: 'author_id'
   has_many :comments, foreign_key: 'author_id'
   has_many :likes, foreign_key: 'author_id'
 
-  validates :name, presence: true
-  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  # validates :name, presence: true
+  # validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```
Comment (Lizdev-05) [inline app/models/user.rb:12]: - [ ] Please make sure you remove commented code from your project to make your code clean and tidy :pray: You should not push commented code to your repository. You can read more about it [here](https://www.freecodecamp.org/news/code-comments-the-good-the-bad-and-the-ugly-be9cc65fbf83/)

File: app/models/user.rb [hunk 3]
```diff
@@ -1,10 +1,14 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (Lizdev-05) [inline app/models/user.rb:3]: - [ ] Please make sure you remove commented code from your project to make your code clean and tidy :pray: You should not push commented code to your repository. You can read more about it [here](https://www.freecodecamp.org/news/code-comments-the-good-the-bad-and-the-ugly-be9cc65fbf83/)

**Review by Lizdev-05 — CHANGES_REQUESTED** [review]
Hi @Naledi-Dikgale and @lugard1 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
:heavy_check_mark: Professional Readme file :+1: 
:heavy_check_mark: Good commit history :+1:
:heavy_check_mark: No linter error :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
```
Comment (KanzaTahreem) [inline app/models/user.rb:3]: - [ ] As suggested by the reviewer [here](https://github.com/Naledi-Dikgale/blog-app/pull/13#discussion_r1274870050), kindly make sure to remove this piece of commented code to make your codebase clean and tidy 📌✨

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @Naledi-Dikgale 

Good job so far on making the changes requested by a previous reviewer 👏! I really appreciate the effort you put in!  **_However, there are few this that still need to be fixed. But you are almost there!_** 💪 🏁

so close


## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ Devise is used
- ✅ Correct Linters
- ✅ Descriptive PR and README file

## Required Changes 🛑

_Please check the comments under the review._

## Optional suggestions ❕

N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**